process/receipts/receiptsHolder.go

```diff
@@ -0,0 +1,67 @@
+package receipts
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-core/data/batch"
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+func newReceiptsHolder() *process.ReceiptsHolder {
+	return &process.ReceiptsHolder{
+		Miniblocks: make([]*block.MiniBlock, 0),
+	}
+}
+
+func marshalReceiptsHolder(holder *process.ReceiptsHolder, marshaller marshal.Marshalizer) ([]byte, error) {
+	receiptsBatch := &batch.Batch{}
+
+	for _, miniBlock := range holder.Miniblocks {
```

Comment:
this will panic if holder is nil

Reply:
Added extra checks (in `receiptsRepository`, the one with exported functions).